Meadowlark-js/packages/meadowlark-core/src/handler/FrontendRequest.ts

```diff
@@ -67,8 +72,29 @@ export function newFrontendRequest(): FrontendRequest {
     traceId: '',
     body: null,
     headers: {},
-    queryStringParameters: {},
+    queryParameters: {},
     stage: '',
     middleware: newFrontendRequestMiddleware(),
   };
 }
+
+/**
+ * Type of function that returns a new FrontendRequest based on the given one, with some values set differently.
+ */
+type FrontendRequestSetter = (f: FrontendRequest) => FrontendRequest;
+
+/**
+ * Creates copy of a FrontendRequest suitable for logging, such as removing non-serializable fields
+ */
+export function frontendRequestForLogging(frontendRequest: FrontendRequest): FrontendRequest {
```

Comment:
The code that cleans up frontendRequest for logging. It creates a new frontendRequest with a matchingMetaEdModel that only has the metaEdName filled in. Serializes fine but kind of verbose, so we may want to just eliminate.

I'm using Ramda lenses here to make the substitution. You can read about them here: https://medium.com/javascript-scene/lenses-b85976cb0534

Reply:
Yes, much too verbose 😆 

Here I deliberately have a bad URL, so there is no matching MetaEd model.

```
2022-09-09 08:50:72 info ParsePathMiddleware.parsePath /v3.3b/ed-fi/dcontentClassDescriptors {"action":"upsert","path":"/v3.3b/ed-fi/dcontentClassDescriptors","traceId":"91088350-073d-49fa-b6be-d75186ad545d","body":"{\r\n  \"codeValue\": \"Presentation\",\r\n  \"shortDescription\": \"Presentation\",\r\n  \"description\": \"Presentation\",\r\n  \"namespace\": \"uri://ed-fi.org/ContentClassDescriptor\"\r\n}","headers":{"user-agent":"vscode-restclient","authorization":"bearer eyJ0eXAiOiJKV1QiLCJhbGciOiJIUzI1NiJ9.eyJpc3MiOiJlZC1maS1tZWFkb3dsYXJrIiwiYXVkIjoibWVhZG93bGFyayIsInJvbGVzIjpbInZlbmRvciJdLCJjbGllbnRfaWQiOiJtZWFkb3dsYXJrX2tleV8xIiwic3ViIjoic3VwZXItZ3JlYXQtU0lTIiwianRpIjoiMTRkZThhODQtMWY5NS00YzE5LWE5NTQtYTAwNzFjNjhhNjQwIiwiaWF0IjoxNjYyNzMxMzYwLCJleHAiOjE2NjI3MzQ5NjB9.8WiF4u6RObZIApPhsvmYGqML05fI76YF7gETzT3AVvg","content-type":"application/json","accept-encoding":"gzip, deflate","content-length":"166","host":"localhost:3000","connection":"close","Host":"localhost:3000"},"queryParameters":{},"stage":"local","middleware":{"security":{"authorizationStrategy":"OWNERSHIP_BASED","clientId":"meadowlark_key_1"},"pathComponents":{"version":"","namespace":"","endpointName":"","resourceId":null},"parsedBody":{},"resourceInfo":{"resourceName":"NoResourceName","isDescriptor":false,"projectName":"NoProjectName","resourceVersion":"0.0.0","namespace":"NoNamespace"},"matchingMetaEdModel":{"type":"unknown","isDeprecated":false,"deprecationReason":"","documentation":"","metaEdName":"NoTopLevelEntity","metaEdId":"","namespace":{"entity":{"unknown":{},"association":{},"associationExtension":{},"associationSubclass":{},"choice":{},"common":{},"commonExtension":{},"commonSubclass":{},"decimalType":{},"descriptor":{},"domain":{},"domainEntity":{},"domainEntityExtension":{},"domainEntitySubclass":{},"enumeration":{},"integerType":{},"interchange":{},"interchangeExtension":{},"mapTypeEnumeration":{},"schoolYearEnumeration":{},"sharedDecimal":{},"sharedInteger":{},"sharedString":{},"stringType":{},"subdomain":{}},"namespaceName":"nonamespace","isExtension":false,"dependencies":[],"projectExtension":"NoNamespace","projectName":"","projectVersion":"","projectDescription":"","extensionEntitySuffix":"Extension","data":{},"config":{},"type":false,"sourceMap":false},"data":{},"config":{},"properties":[],"identityProperties":[],"queryableFields":[],"typeHumanizedName":"","allowPrimaryKeyUpdates":false,"baseEntityName":"","baseEntityNamespaceName":"","baseEntity":null,"extendedBy":[],"subclassedBy":[],"outReferences":[],"inReferences":[],"outReferencePaths":[],"outReferenceEntitiesMap":{},"outReferenceEntityEndpointsMap":{},"sourceMap":{"type":{"line":0,"column":0,"tokenText":"NoSourceMap"},"isDeprecated":{"line":0,"column":0,"tokenText":"NoSourceMap"},"deprecationReason":{"line":0,"column":0,"tokenText":"NoSourceMap"},"documentation":{"line":0,"column":0,"tokenText":"NoSourceMap"},"metaEdName":{"line":0,"column":0,"tokenText":"NoSourceMap"},"metaEdId":{"line":0,"column":0,"tokenText":"NoSourceMap"},"properties":[],"identityProperties":[],"queryableFields":[],"allowPrimaryKeyUpdates":{"line":0,"column":0,"tokenText":"NoSourceMap"},"baseEntityName":{"line":0,"column":0,"tokenText":"NoSourceMap"},"baseEntityNamespaceName":{"line":0,"column":0,"tokenText":"NoSourceMap"},"baseEntity":{"line":0,"column":0,"tokenText":"NoSourceMap"}}},"documentInfo":{"documentIdentity":{},"superclassInfo":null,"documentReferences":[],"descriptorReferences":[]},"headerMetadata":{},"validateResources":true}}
```

And here I've fixed it:

```
2022-09-09 08:55:66 info ValidateQueryMiddleware.queryValidation /v3.3b/ed-fi/contentClassDescriptors {"action":"query","path":"/v3.3b/ed-fi/contentClassDescriptors","traceId":"0b5e4df7-7843-49a9-a963-6d0224a81726","body":null,"headers":{"user-agent":"vscode-restclient","authorization":"bearer eyJ0eXAiOiJKV1QiLCJhbGciOiJIUzI1NiJ9.eyJpc3MiOiJlZC1maS1tZWFkb3dsYXJrIiwiYXVkIjoibWVhZG93bGFyayIsInJvbGVzIjpbInZlbmRvciJdLCJjbGllbnRfaWQiOiJtZWFkb3dsYXJrX2tleV8xIiwic3ViIjoic3VwZXItZ3JlYXQtU0lTIiwianRpIjoiMTRkZThhODQtMWY5NS00YzE5LWE5NTQtYTAwNzFjNjhhNjQwIiwiaWF0IjoxNjYyNzMxMzYwLCJleHAiOjE2NjI3MzQ5NjB9.8WiF4u6RObZIApPhsvmYGqML05fI76YF7gETzT3AVvg","accept-encoding":"gzip, deflate","host":"localhost:3000","connection":"close","Host":"localhost:3000"},"queryParameters":{},"stage":"local","middleware":{"security":{"authorizationStrategy":"OWNERSHIP_BASED","clientId":"meadowlark_key_1"},"pathComponents":{"version":"v3.3b","namespace":"ed-fi","endpointName":"contentClassDescriptors"},"parsedBody":{},"resourceInfo":{"projectName":"Ed-Fi","resourceVersion":"3.3.1-b","resourceName":"ContentClass","isDescriptor":true},"matchingMetaEdModel":{"type":"unknown","isDeprecated":false,"deprecationReason":"","documentation":"","metaEdName":"ContentClass","metaEdId":"","namespace":{"entity":{"unknown":{},"association":{},"associationExtension":{},"associationSubclass":{},"choice":{},"common":{},"commonExtension":{},"commonSubclass":{},"decimalType":{},"descriptor":{},"domain":{},"domainEntity":{},"domainEntityExtension":{},"domainEntitySubclass":{},"enumeration":{},"integerType":{},"interchange":{},"interchangeExtension":{},"mapTypeEnumeration":{},"schoolYearEnumeration":{},"sharedDecimal":{},"sharedInteger":{},"sharedString":{},"stringType":{},"subdomain":{}},"namespaceName":"nonamespace","isExtension":false,"dependencies":[],"projectExtension":"NoNamespace","projectName":"","projectVersion":"","projectDescription":"","extensionEntitySuffix":"Extension","data":{},"config":{},"type":false,"sourceMap":false},"data":{},"config":{},"properties":[],"identityProperties":[],"queryableFields":[],"typeHumanizedName":"","allowPrimaryKeyUpdates":false,"baseEntityName":"","baseEntityNamespaceName":"","baseEntity":null,"extendedBy":[],"subclassedBy":[],"outReferences":[],"inReferences":[],"outReferencePaths":[],"outReferenceEntitiesMap":{},"outReferenceEntityEndpointsMap":{},"sourceMap":{"type":{"line":0,"column":0,"tokenText":"NoSourceMap"},"isDeprecated":{"line":0,"column":0,"tokenText":"NoSourceMap"},"deprecationReason":{"line":0,"column":0,"tokenText":"NoSourceMap"},"documentation":{"line":0,"column":0,"tokenText":"NoSourceMap"},"metaEdName":{"line":0,"column":0,"tokenText":"NoSourceMap"},"metaEdId":{"line":0,"column":0,"tokenText":"NoSourceMap"},"properties":[],"identityProperties":[],"queryableFields":[],"allowPrimaryKeyUpdates":{"line":0,"column":0,"tokenText":"NoSourceMap"},"baseEntityName":{"line":0,"column":0,"tokenText":"NoSourceMap"},"baseEntityNamespaceName":{"line":0,"column":0,"tokenText":"NoSourceMap"},"baseEntity":{"line":0,"column":0,"tokenText":"NoSourceMap"}}},"documentInfo":{"documentIdentity":{},"superclassInfo":null,"documentReferences":[],"descriptorReferences":[]},"headerMetadata":{"X-MetaEd-Project-Name":"Ed-Fi","X-MetaEd-Project-Version":"3.3.1-b","X-MetaEd-Project-Package-Name":"@edfi/ed-fi-model-3.3b"},"validateResources":true}}
```

There is almost no information that we actually need in there, to be frank. Please change so that the metaed model information is only logged once, right after it is found, with DEBUG level instead of INFO. I'm not sure there's really value in having the entire metaed model document in the log. I can see that there _is_ value in having some basic information, especially when debugging year issues. Would it be sufficient to just log a few things, such as metaEdName, RoleName, and isDescriptor?